frontend/src/metabase-lib/expressions/resolver.js

```diff
@@ -42,7 +42,13 @@ const isCompatible = (a, b) => {
   if (a === b) {
     return true;
   }
-  if (a === "expression" && (b === "number" || b === "string")) {
+  if (a === "datetime" && b === "string") {
+    return true;
+  }
```

Comment:
I wonder if we should document this a little? Why is a datetime arg compatible with a string?

Reply:
Yes, thank you. On this.

Reply:
@calherries, done.

Is this kind of what you meant?

I agree it's important to document this addition, especially as these expression types are quite dispersed in the code.

At the same time, the reason "datetime" is compatible in theses cases doesn't seem complicated.

Reply:
> At the same time, the reason "datetime" is compatible in theses cases doesn't seem complicated.

Yes, not complicated, but still potentially bewildering. We should also document it as something to fix in the future - by doing this we are allowing general string columns to be arguments to datetime functions, which is definitely not valid on the backend. Only literals are allowed.

> // datetime type is a subset of "string" type,

I would say this is misleading. `datetime` typed columns, for example, aren't allowed to be arguments to string functions. This is how I understand it: `a` is the type of the argument expected, as defined in `MBQL_CLAUSES`, and `b` is the inferred type of the argument. So `(a === "datetime" && b === "string")` is saying if b is a string, then it can be an arg to a function that expects a datetime argument.

Reply:
Updated